### PR TITLE
Artificer Content

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

**TL;DR: This PR adds lots of new smithable/craftable items for artificers along with some artificer-related QoL changes. Added smoke bombs, craftable scomstones, and lots of other little things.**

More details below.

<details>

**QoL changes:**
- Artificer engineering skill raised to master. This is their whole thing - they should be good at it.
- Gave artificer journeyman blacksmithing for making basic things like lockpicks, rings, and lampterns. They still don't get any weaponsmithing/armorsmithing skills so this shouldn't intrude on the blacksmith's job too much.
- Engineering anvil recipes now list components needed for each recipe.
- Bronze lampterns, keys, and locks are now made in batches of 3.

**New Engineering Craftables:**
- Scomstones/serfstones are now smithable with high engineering skill; serfstones require a steel bar - scomstones require a gold bar.
- Talkstones are now smithable with high engineering skill; requires a gold bar and a dorpel.
- Dwarven music boxes can be made with legendary engineering, a riddle of steel, and 2 cogs.
- Wall SCOMs are now made with engineering skill and 1 bronze cog + 1 iron bar.
- Poisoned arrows/bolts and pyroclastic arrows/bolts both now require high engineering skill to craft - specialized arrows/bolts are admittedly pretty strong but now require an artificer to make them.
- The mining helmets with attached lampterns that underdwellers spawn with can now be made with engineering skill.

**Entirely New Items:**
- New *reinforced steel mining helmet* variant; requires someone with high engineering skill to 'upgrade' a kettleman helmet. It's essentially just a kettleman helmet with a lamptern attached to it - which saves you potentially having to carry a lamptern in your belt slot.
- New *enchanted signal horn* item able to be created with high engineering skill. Functions like the message spell but without the ability to anonymize yourself - allows you to send one message to another player of your choosing from any distance. The horn is destroyed upon successful use; this is a one-time-use item.
- New *smoke bombs* - one time use items made with bronze and iron that can be tossed onto the ground to activate them; will cause the user to vanish in a puff of smoke and turn invisible for a brief few seconds - great escape tool for antagonist characters!

</details>

## Why It's Good For The Game

Artificer desperately needs content - the role has basically nothing going for it. This is an attempt to really cement them as builders and arcane craftsmen by giving them a few more unique crafting recipes and allowing them to craft things like scomstones, talkstones, specialized arrows, and a few other nicher items.

I'll probably add more stuff later in another PR; but I'm happy enough with this.

### CODE REVIEW

**Comment by TheEternalFlame** [comment]
I also think they could do with maybe journeyman crossbow skills

**Comment by radarseas** [comment]
could you throw them the `trained smith` trait that blacksmtihs get. So if they grind hard enough they can't be replaced by a pilgrim smithy.

it might also be worth moving scom construction from arcayne to just engineering alone, literally nobody knows you can make scoms since they need an ingot and a rat, _**a rat**_ of all things.
